python/lsst/rubintv/production/rubinTv.py

```diff
@@ -825,3 +850,201 @@ def beat(self, ensure=False, customFlatlinePeriod=None):
+        """
+        refs = self.butler.registry.queryDatasets(refcatName, dataId=dataId).expanded()
+        refs = list(refs)
+        handles = [dafButler.DeferredDatasetHandle(butler=self.butler, ref=ref, parameters=None)
+                   for ref in refs]
+        dataIds = [ref.dataId for ref in refs]
+
+        loader = ReferenceObjectLoader(
+            dataIds,
+            handles,
+            name=refcatName,
+            log=self.log,
+            config=config
+        )
+        return loader
+
+    def callback(self, dataId, **kwargs):
+        """Method called on each new dataId as it is found in the repo.
+
+        Runs on the quickLookExp and writes shards with various measured
+        quantities, as calculated by the CharacterizeImageTask and
+        CalibrateTask.
+        """
+        try:
+            tStart = time.time()
+
+            self.log.info(f'Running Image Characterization for {dataId}')
+            exp = _waitForDataProduct(self.butler, self.dataProduct, dataId, self.log)
+
+            if not exp:
+                raise RuntimeError(f'Failed to get {self.dataProduct} for {dataId}')
+
+            if isDispersedExp(exp):
+                self.log.info(f'Skipping dispersed image: {dataId}')
+                return
+
+            self.defineVisit(dataId)
+            visitDataId = self.getVisitDataId(dataId)
+
+            loader = self._getRefObjLoader(self.calibrate.config.connections.astromRefCat, visitDataId,
+                                           config=self.calibrate.config.astromRefObjLoader)
+            self.calibrate.astrometry.setRefObjLoader(loader)
+            loader = self._getRefObjLoader(self.calibrate.config.connections.photoRefCat, visitDataId,
+                                           config=self.calibrate.config.photoRefObjLoader)
+            self.calibrate.photoCal.match.setRefObjLoader(loader)
+
+            charRes = self.charImage.run(exp)
+            tCharacterize = time.time()
+            self.log.info(f"Ran characterizeImageTask in {tCharacterize-tStart:.2f} seconds")
+
+            nSources = len(charRes.sourceCat)
+            dayObs = butlerUtils.getDayObs(dataId)
+            seqNum = butlerUtils.getSeqNum(dataId)
+            outputDict = {"50-sigma source count": nSources}
+            mdDict = {seqNum: outputDict}
+            writeMetadataShard(self.shardsDir, dayObs, mdDict)
+
+            calibrateRes = self.calibrate.run(charRes.exposure,
+                                              background=charRes.background,
+                                              icSourceCat=charRes.sourceCat)
+            tCalibrate = time.time()
+            self.log.info(f"Ran calibrateTask in {tCalibrate-tCharacterize:.2f} seconds")
+
+            summaryStats = calibrateRes.outputExposure.getInfo().getSummaryStats()
+            pixToArcseconds = calibrateRes.outputExposure.getWcs().getPixelScale().asArcseconds()
```

Comment:
I think we should grab the wcs above before the fit and report the change in the wcs here too, including the change in the rotation angle. Can be another ticket if you don't want to do it on this one, but it would be nice to add at this point I think.

Reply:
Agreed, but it's not going to get merged tonight if I do that now.